### PR TITLE
Add support for camel property placeholders (& resolution)

### LINKED ISSUE #385: CamelConnector:  Support supplying property values for property placeholders
Apache Camel routes support the notion of property placeholders, indicated by double curly brackets, _e.g._, `{{placeholder}}`. At runtime, values can be supplied for these, and the Camel runtime will fill them in in the route specification.

Making these available allows for better reuse of existing route descriptions.

### PR DESCRIPTION
**Pull request by fhcarter**
Fixes #385

Apache Camel has support for including property placeholders in routes so that the route can be specialized at the time of invocation.  These are done by including the property placeholder name in double curly brackets (_e.g._, `{{somePropertyName}}`).  We don't really have to deal with these -- they are all handled by the camel mechanism.

However, for that mechanism to work, we must provide a way to pass values in.  We do this by adding a `propertyValues` section to a source configuration for the Camel connector.  When so provided, this must be a JSON object with values for properties keyed by the property name.

When this is provided, we'll initialize the `PropertiesComponent` in Camel with these values and let camel do its job.

Add tests & documentation (the readme).  Turns out, the documentation for component properties (used in component configuration) wasn't added, so added that, too.

### CODE REVIEW

**Comment by fhcarter** [comment]
[Jenkins run](https://jenkins.vantiq.com/jenkins/job/vantiq-extension-sources-branch/133/)

**Comment by fhcarter** [comment]
This is particularly useful in making use of prebuilt routes from others.

**Review by jcguion — APPROVED** [review]
LGTM

File: camelConnector/src/main/java/io/vantiq/extsrc/camelconn/connector/CamelHandleConfiguration.java [hunk 1]
```diff
@@ -132,6 +134,36 @@ public void handleMessage(ExtensionServiceMessage message) {
                       ROUTES_LIST, ROUTES_FORMAT);
         }
```
[the remark this answers is not in the document]
Comment (fhcarter) [reply camelConnector/src/main/java/io/vantiq/extsrc/camelconn/connector/CamelHandleConfiguration.java:135]: Probably -- Camel may be able to guess correctly, but I'll add the change.

File: camelConnector/README.md [hunk 2]
```diff
@@ -132,11 +132,48 @@ The Configuration document may look similar to the following example:
 
 ### Options Available for camelRuntime
 
-*   **appName**: Optional -- name of the app.  If not provided, default to _camelConnectorApp_.
-*   **routesDocument** or **routesList** and **routesFormat**: Required. This is list of Camel routes to run.
-    * **routesDocument** -- the name of a Vantiq Document containing the XML or YAML definition of the routes to run. The format of the document (XML or YAML) will be determined from the file name (_e.g._, `routes.xml`)
-    * **routesList** and **routesFormat** -- the XML or YAML specification of the Camel routes to run.  The **routesList** property should contain the appropriate text, and the **routesFormat** should contain either `xml` or `yml` indicating the style used to specify the routes.
-    * *Note*: You must specify either the **routesDocument** or **routesList** _AND_ **routesFormat** properties, but not both.
+* **appName**: Optional -- name of the app.  If not provided, default to _camelConnectorApp_.
+* **routesDocument** or **routesList** and **routesFormat**: Required. This is list of Camel routes to run.
```
[the remark this answers is not in the document]
Comment (fhcarter) [reply camelConnector/README.md:136]: overzealous cleanup :-)

[the remark this answers is not in the document]
Comment (fhcarter) [reply camelConnector/README.md:136]: Getting the embedded ```json ... ``` stuff to work was heavily indent dependent, so probably lost it a bit there.

**Pull request merged**